debian/appscale_install_functions.sh

```diff
@@ -573,3 +585,79 @@ upgradepip()
             ;;
     esac
 }
+
+buildmapreduce()
+{
+    # This function compiles the AppScale compatible MapReduce, Pipeline and GCS client libraries, builds the
+    # jars and places them under appscale/ext/java.
+
+    # Clone the AppScale repositories for the libaries.
+    git clone -b ${APPSCALE_GCS_BRANCH} ${APPSCALE_GCS_CLIENT} appscale-gcs-client
+    git clone -b ${APPSCALE_PL_BRANCH} ${APPSCALE_PIPELINE} appscale-pipelines
+    git clone -b ${APPSCALE_MR_BRANCH} ${APPSCALE_MAPREDUCE} appscale-mapreduce
+
+    echo -n "Building AppScale Google Cloud Storage Client jar ..."
+    if ! (cd appscale-gcs-client/java; mvn compile); then
```

Comment:
I think it's best practice to separate commands with `&&` here. That way if the directory doesn't exist for some reason, the `mvn compile` won't run.